### PR TITLE
fix(DTFS2-6655): createBssEwcsDeal cypress command - individual deals

### PR DESCRIPTION
**Pull request by vamshidharukef**
## Introduction :pencil2:
For some of the e2e tests after validation is put on users, some tests fail as users with role maker cannot access end point `http://localhost:5001/v1/users`

## Resolution :heavy_check_mark:
Create cy.createBssApplicationReadyForReview() for creating BSS/EWCS deals through UI
Update commnds.js to use cy.createBssApplicationReadyForReview

### CODE REVIEW

File: e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  bankDetails.bankDealName().type('BSS Deal');
+  bankDetails.submit().click();
+  contract.aboutSupplierDetailsLink().click();
+  aboutSupplier.supplierType().select('Exporter');
+  aboutSupplier.supplierCompaniesHouseRegistrationNumber().type('12345678');
+  aboutSupplier.supplierSearchCompaniesHouse().click();
+  aboutSupplier.supplierAddress().country().select('United Kingdom');
+  aboutSupplier.supplierCorrespondenceAddressSame().click();
+  aboutSupplier.smeTypeSmall().click();
+  aboutSupplier.supplyContractDescription().type('Supply Contract Description');
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js:37]: For all `.type` instances in this command, can we please add `{ delay: 0 }`? E.g:

```js
aboutSupplier.supplyContractDescription().type('Supply Contract Description', { delay: 0 });
```

This minimises the time it takes to enter text as much as possible. When this command is consumed in all the relevant E2E tests, this change will significantly reduce the time it takes to execute 👍 

We need to create a generic `keyboardInput` type cypress command, but this is one for another day.

File: e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  bondFinancialDetails.currencySameAsSupplyContractCurrencyYesInput().click();
+  bondFinancialDetails.riskMarginFeeInput().type('10');
+  bondFinancialDetails.coveredPercentageInput().type('80');
+  bondFinancialDetails.submit().click();
+  bondFeeDetails.feeTypeAtMaturityInput().click();
+  bondFeeDetails.dayCountBasis365Input().click();
+  bondFeeDetails.saveGoBackButton().click();
+  contract.proceedToReview().click();
+  contractReadyForReview.comments().type('Ready for checkers approval');
+  contractReadyForReview.readyForCheckersApproval().click();
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js:78]: Can we add an empty line after each section please?

File: e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+const bondDetails = require('../../e2e/pages/bondDetails');
+const bondFinancialDetails = require('../../e2e/pages/bondFinancialDetails');
+const bondFeeDetails = require('../../e2e/pages/bondFeeDetails');
+const contractReadyForReview = require('../../e2e/pages/contract/contract-readyForReview');
+const login = require('./login');
+const MOCK_USERS = require('../../../../e2e-fixtures/index');
+
+const { BANK1_MAKER1 } = MOCK_USERS;
+
+const createBssApplicationReadyForCheck = () => {
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js:20]: documentation?

File: e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+const contract = require('../../e2e/pages/contract');
+const aboutSupplier = require('../../e2e/pages/contract/about/about-supplier');
+const aboutBuyer = require('../../e2e/pages/contract/about/about-buyer');
+const aboutFinancial = require('../../e2e/pages/contract/about/about-financial');
+const eligibilityCriteria = require('../../e2e/pages/eligibilityCriteria');
+const eligibilityPreview = require('../../e2e/pages/eligibilityPreview');
+const bondDetails = require('../../e2e/pages/bondDetails');
+const bondFinancialDetails = require('../../e2e/pages/bondFinancialDetails');
+const bondFeeDetails = require('../../e2e/pages/bondFeeDetails');
+const contractReadyForReview = require('../../e2e/pages/contract/contract-readyForReview');
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js:14]: For all the pages imports, could do this instead with 1x import/require 😎 

```js
const {
  dashboard,
  selectScheme,
  beforeYouStart,
  bankDetails,
  contract,
  aboutSupplier,
  // ...
} = require('../../e2e/pages');
```

**Review by ttbarnes — CHANGES_REQUESTED** [review]

File: e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js (hunk 1 above)
Comment (vamshidharukef) [reply e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js:37]: Added `{ delay: 0 }` for `.type` instances.

File: e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js (hunk 2 above)
Comment (vamshidharukef) [reply e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js:78]: Added empty line with comment for each section

File: e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js (hunk 3 above)
Comment (vamshidharukef) [reply e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js:20]: Sorry, what documentation is required?

File: e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js (hunk 4 above)
Comment (vamshidharukef) [reply e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js:14]: Imported as requested.

File: e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js (hunk 3 above)
Comment (ttbarnes) [reply e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js:21]: Something like this?

```js
/**
 * Create a BSS/EWCS deal via the UI with a "ready for check" status
 */
```

Does the command need to return the deal?

Comment (ttbarnes) [reply e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js:21]: Also, should we call this `createBssDealReadyForCheck` instead of `createBssApplicationReadyForCheck`?

Comment (vamshidharukef) [reply e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js:21]: Added documentation and changed from `createBssApplicationReadyForCheck` to `createBssDealReadyForCheck`.

File: e2e-tests/portal/cypress/support/commands.js [hunk 5]
```diff
@@ -83,6 +83,7 @@ Cypress.Commands.add('renameDeal', require('./portal/renameDeal'));
 Cypress.Commands.add('resetPortalUserStatusAndNumberOfSignInLinks', require('./portal/resetPortalUserStatusAndNumberOfSignInLinks'));
 Cypress.Commands.add('userSetPassword', require('./portal/userSetPassword'));
 Cypress.Commands.add('disablePortalUserByUsername', require('./portal/disablePortalUserByUsername'));
+Cypress.Commands.add('createBssApplicationReadyForCheck', require('./portal/createBssDealReadyForCheck'));
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/support/commands.js:86]: Should be `createBssDeal` instead of `createBssApplicationReadyForCheck` please

File: e2e-tests/portal/cypress/support/portal/createBssDealReadyForCheck.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+} = require('../../e2e/pages');
+const { aboutSupplier, aboutBuyer, aboutFinancial } = require('../../e2e/pages/contract');
+const login = require('./login');
+const MOCK_USERS = require('../../../../e2e-fixtures/index');
+
+const { BANK1_MAKER1 } = MOCK_USERS;
+
+/** * Create a BSS/EWCS deal via the UI with a "ready for check" status */
+
+const createBssDealReadyForCheck = () => {
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/support/portal/createBssDealReadyForCheck.js:23]: Can we make the "Ready for check" part conditional?

This way the default is to create a deal in a "draft" state, which is the most minimal and will reduce E2E time. E.g:

```js
/**
 * Create a BSS/EWCS deal via the UI
 * Conditionally complete all fields and submit,
 * so that the deal has a a "ready for check" status
 * @param {Boolean} readyForCheck
 */
const createBssDeal = ({ readyForCheck = false }) => {
  // etc...
  // Fill in bank deal id, bank name

  if (readyForCheck) {
    // do all other steps
  }
};
```

Then, when replacing instances of `cy.insertOneDeal`, depending on the conditions used throughout for the existing command, we can also adapt this to do other things e.g:

```
const createBssDeal = ({
  readyForCheck = false,
  unissuedFacilities = false,
}) => {
  if (readyForCheck) {
    // do all other steps
  }

  if (unissuedFacilities) {
    // created
  }
};
```

**Review by ttbarnes — CHANGES_REQUESTED** [review]

File: e2e-tests/portal/cypress/support/commands.js (hunk 5 above)
Comment (vamshidharukef) [reply e2e-tests/portal/cypress/support/commands.js:86]: Changed to `createBssDeal`

File: e2e-tests/portal/cypress/support/portal/createBssDealReadyForCheck.js (hunk 6 above)
Comment (vamshidharukef) [reply e2e-tests/portal/cypress/support/portal/createBssDealReadyForCheck.js:23]: Added "`Ready for check`" part conditional

File: e2e-tests/portal/cypress/e2e/journeys/admin/restrict-access/restrict-access.spec.js [hunk 7]
```diff
@@ -13,17 +12,13 @@ context('Only allow authorised users to access admin pages', () => {
...
-
       cy.deleteDeals(ADMIN);
-      cy.insertOneDeal(aDealInStatus('Draft'), BANK1_MAKER1).then((insertedDeal) => {
-        deal = insertedDeal;
-      });
+      cy.createBssDeal({});
     });
 
     it('allows read only user with all bank access to view deal', () => {
       cy.loginGoToDealPage(ADMIN, deal);
```
Comment (abhi-markan) [inline e2e-tests/portal/cypress/e2e/journeys/admin/restrict-access/restrict-access.spec.js:20]: We are login as an `ADMIN` whereas the assertion depicts as a `read-only` user?

File: e2e-tests/portal/cypress/support/commands.js [hunk 8]
```diff
@@ -83,6 +83,7 @@ Cypress.Commands.add('renameDeal', require('./portal/renameDeal'));
 Cypress.Commands.add('resetPortalUserStatusAndNumberOfSignInLinks', require('./portal/resetPortalUserStatusAndNumberOfSignInLinks'));
 Cypress.Commands.add('userSetPassword', require('./portal/userSetPassword'));
 Cypress.Commands.add('disablePortalUserByUsername', require('./portal/disablePortalUserByUsername'));
+Cypress.Commands.add('createBssDeal', require('./portal/createBssDeal'));
```
Comment (abhi-markan) [inline e2e-tests/portal/cypress/support/commands.js:86]: Can this be `createBssEwcsDeal` please?

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: e2e-tests/portal/cypress/support/commands.js (hunk 5 above)
Comment (abhi-markan) [reply e2e-tests/portal/cypress/support/commands.js]: I was wondering if it should be `createBssEwcsDeal`?

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filter-by-keyword.spec.js [hunk 9]
```diff
@@ -87,10 +78,6 @@ context('Dashboard Deals filters - filter by keyword', () => {
     it(`renders only deals that have ${MOCK_KEYWORD} in a field`, () => {
       const ALL_KEYWORD_DEALS = ALL_DEALS.filter(({ exporter }) => exporter.companyName === MOCK_KEYWORD);
       dashboardDeals.rows().should('have.length', ALL_KEYWORD_DEALS.length);
-
-      const firstDraftDeal = ALL_KEYWORD_DEALS[0];
-
-      dashboardDeals.row.exporter(firstDraftDeal._id).should('have.text', MOCK_KEYWORD);
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filter-by-keyword.spec.js:93]: Can we re-add this test please

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filter-by-notice-type.spec.js [hunk 10]
```diff
@@ -79,10 +77,6 @@ context('Dashboard Deals filters - filter by submissionType/noticeType', () => {
     it('renders only MIA deals', () => {
       const ALL_MIA_DEALS = ALL_DEALS.filter(({ submissionType }) => submissionType === CONSTANTS.DEALS.SUBMISSION_TYPE.MIA);
       dashboardDeals.rows().should('have.length', ALL_MIA_DEALS.length);
-
-      const firstMiaDeal = ALL_MIA_DEALS[0];
-
-      dashboardDeals.row.type(firstMiaDeal._id).should('have.text', CONSTANTS.DEALS.SUBMISSION_TYPE.MIA);
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filter-by-notice-type.spec.js:85]: Same as above

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filter-by-product.spec.js [hunk 11]
```diff
@@ -133,10 +127,6 @@ context('Dashboard Deals filters - filter by dealType/product', () => {
     it('renders only GEF deals', () => {
       const ALL_GEF_DEALS = ALL_DEALS.filter(({ dealType }) => dealType === CONSTANTS.DEALS.DEAL_TYPE.GEF);
       dashboardDeals.rows().should('have.length', ALL_GEF_DEALS.length);
-
-      const firstGefDeal = ALL_GEF_DEALS[0];
-
-      dashboardDeals.row.product(firstGefDeal._id).should('have.text', CONSTANTS.DEALS.DEAL_TYPE.GEF);
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filter-by-product.spec.js:139]: Can we re-add the removed tests in this file

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-facilities/dashboard-facilities-filters.spec.js [hunk 12]
```diff
@@ -15,7 +15,9 @@ context('Dashboard Deals filters', () => {
     cy.deleteGefApplications(ADMIN);
     cy.deleteDeals(ADMIN);
 
-    cy.insertOneDeal(BSS_DEAL_DRAFT, BANK1_MAKER1);
+    cy.createBssDeal({ readyForCheck: true }).then((deal) => {
+      ALL_FACILITIES.push(deal._id);
+    });
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-facilities/dashboard-facilities-filters.spec.js:20]: This won't work as `createBssDeal` doesn't return a deal ID

File: e2e-tests/portal/cypress/e2e/journeys/maker/abandon-a-deal.spec.js [hunk 13]
```diff
@@ -2,18 +2,16 @@ const { contract, contractDelete, defaults } = require('../../pages');
...
 context('A maker selects to abandon a contract from the view-contract page', () => {
   let deal;
 
   before(() => {
-    const aDealInStatus = (status) => twentyOneDeals.filter((aDeal) => status === aDeal.status)[0];
-
     cy.deleteDeals(ADMIN);
-    cy.insertOneDeal(aDealInStatus('Draft'), BANK1_MAKER1).then((insertedDeal) => {
+
+    cy.createBssDeal({ readyForCheck: true }).then((insertedDeal) => {
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/maker/abandon-a-deal.spec.js:14]: This should be `cy.createBssDeal({});` so that the deal has a `DRAFT` status like the previous test

File: e2e-tests/portal/cypress/e2e/journeys/maker-bond/bond.spec.js [hunk 14]
```diff
@@ -23,7 +23,7 @@ context('Add a Bond to a Deal', () => {
 
   beforeEach(() => {
     cy.deleteDeals(ADMIN);
-    cy.insertOneDeal(MOCK_DEAL, BANK1_MAKER1).then((insertedDeal) => {
+    cy.createBssDeal({ readyForCheck: true }).then((insertedDeal) => {
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/maker-bond/bond.spec.js:26]: I think this should be `cy.createBssDeal({});` so that the deal has a `DRAFT` status

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-panel-filter-remove.spec.js [hunk 15]
```diff
@@ -15,7 +15,7 @@ context('Dashboard Deals - panel selected filters - remove a filter', () => {
     cy.deleteGefApplications(ADMIN);
     cy.deleteDeals(ADMIN);
 
-    cy.insertOneDeal(BSS_DEAL_DRAFT, BANK1_MAKER1).then((deal) => {
+    cy.createBssDeal({ readyForCheck: true }).then((deal) => {
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-panel-filter-remove.spec.js:18]: This should be `cy.createBssDeal({});` so that the deal has a `DRAFT` status like the previous test

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-main-container-filter-remove.spec.js [hunk 16]
```diff
@@ -15,7 +15,7 @@ context('Dashboard Deals - main container selected filters - remove a filter', (
     cy.deleteGefApplications(ADMIN);
     cy.deleteDeals(ADMIN);
 
-    cy.insertOneDeal(BSS_DEAL_DRAFT, BANK1_MAKER1).then((deal) => {
+    cy.createBssDeal({ readyForCheck: true }).then((deal) => {
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-main-container-filter-remove.spec.js:18]: This should be `cy.createBssDeal({});` so that the deal has a `DRAFT` status like the previous test

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filters.spec.js [hunk 17]
```diff
@@ -15,7 +15,7 @@ context('Dashboard Deals filters', () => {
     cy.deleteGefApplications(ADMIN);
     cy.deleteDeals(ADMIN);
 
-    cy.insertOneDeal(BSS_DEAL_DRAFT, BANK1_MAKER1).then((deal) => {
+    cy.createBssDeal({ readyForCheck: true }).then((deal) => {
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filters.spec.js:18]: This should be `cy.createBssDeal({});` so that the deal has a `DRAFT` status like the previous test

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filters-reset-after-navigating-away.spec.js [hunk 18]
```diff
@@ -11,7 +10,7 @@ context('Dashboard Deals filters - reset after applying and navigating away', ()
     cy.deleteGefApplications(ADMIN);
     cy.deleteDeals(ADMIN);
 
-    cy.insertOneDeal(BSS_DEAL_MIA, BANK1_MAKER1);
+    cy.createBssDeal({ readyForCheck: true });
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filters-reset-after-navigating-away.spec.js:13]: I think this can just be `cy.createBssDeal({})`

File: e2e-tests/portal/cypress/support/portal/createBssDeal.js [hunk 19]
```diff
@@ -0,0 +1,125 @@
...
+  contractAboutBuyer,
+  contractAboutFinancial,
+} = require('../../e2e/pages');
+
+const login = require('./login');
+const MOCK_USERS = require('../../../../e2e-fixtures/index');
+
+const { BANK1_MAKER1 } = MOCK_USERS;
+
+/** * Create a BSS/EWCS deal via the UI with a "ready for check" status */
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/support/portal/createBssDeal.js:25]: Need to add params documentation

**Review by ttbarnes — CHANGES_REQUESTED** [review]

File: e2e-tests/portal/cypress/e2e/journeys/admin/restrict-access/restrict-access.spec.js [hunk 20]
```diff
@@ -13,17 +12,13 @@ context('Only allow authorised users to access admin pages', () => {
...
-      cy.url().should('eq', relative(`/contract/${deal._id}`));
+      cy.url().should('eq', relative(`/dashboard/deals/0`));
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/admin/restrict-access/restrict-access.spec.js:21]: Need to update this test - it was previously testing the contract/deal URL, but now its testing the dashboard URL

File: e2e-tests/portal/cypress/e2e/journeys/checker/cannot-submit-a-deal-with-newly-issued-facilities-that-have-cover-start-dates-before-min-submission-date/cannot-submit-a-deal-with-newly-issued-facilities-that-have-cover-start-dates-before-min-submission-date.spec.js [hunk 21]
```diff
@@ -1,29 +1,24 @@
...
-        deal = insertedDeal;
-        dealId = deal._id;
-      });
+      cy.createBssDeal({ readyForCheck: true });
     });
 
     it('should throw error and not submit or redirect', () => {
       cy.login(BANK1_CHECKER1);
-      pages.contract.visit(deal);
+      pages.dashboardDeals.rowIndex.link(1).click();
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/checker/cannot-submit-a-deal-with-newly-issued-facilities-that-have-cover-start-dates-before-min-submission-date/cannot-submit-a-deal-with-newly-issued-facilities-that-have-cover-start-dates-before-min-submission-date.spec.js:21]: We can update the dashboard to have a default row index

Comment (ttbarnes) [reply e2e-tests/portal/cypress/e2e/journeys/checker/cannot-submit-a-deal-with-newly-issued-facilities-that-have-cover-start-dates-before-min-submission-date/cannot-submit-a-deal-with-newly-issued-facilities-that-have-cover-start-dates-before-min-submission-date.spec.js:21]: I've updated this. I do however think we should create a `clickFirstDashboardDealLink` command or something like that so it's clearer and doesn't require importing in individual tests/commands

File: e2e-tests/portal/cypress/support/portal/createBssDeal.js (hunk 19 above)
Comment (ttbarnes) [reply e2e-tests/portal/cypress/support/portal/createBssDeal.js:25]: I've updated the documentation

**Review by Mihail1Ynakiev — COMMENTED** [review]
Passed bank of Cyprus

**Review by Mihail1Ynakiev — COMMENTED** [review]
image

File: portal/templates/dashboard/_macros/dashboard-deals-table.njk [hunk 22]
```diff
@@ -36,28 +37,31 @@
...
           text: deal.submissionType or "-",
-          attributes: { "data-cy": "deal__submissionType--" + deal._id }
+          attributes: { "data-cy": "deal__submissionType--" + deal._id + " deal__submissionType--index--" + loop.index }
         },
         {
           text: deal.status or "-",
-          attributes: { "data-cy": "deal__status--" + deal._id }
+          attributes: {
+            "data-cy": "deal__status--" + deal._id,
+            "id": "deal__status--index--" + loop.index
```
Comment (ttbarnes) [inline portal/templates/dashboard/_macros/dashboard-deals-table.njk:58]: Need to see if there is a better way of doing this, instead of adding an ID.

Otherwise, perhaps this will have to be temporary until the E2E tests for "multiple BSS/EWCS deals" (dashboard tests), are updated to work without IDs. Then, we can simply replace the data-cys with deal IDs, with indexes.

File: e2e-tests/portal/cypress/support/commands.js (hunk 8 above)
Comment (ttbarnes) [reply e2e-tests/portal/cypress/support/commands.js]: Have updated this to be `createBssEwcsDeal`

File: e2e-tests/portal/cypress/support/commands.js (hunk 5 above)
Comment (ttbarnes) [reply e2e-tests/portal/cypress/support/commands.js]: Resolving, has been renamed.

File: e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js (hunk 1 above)
Comment (ttbarnes) [reply e2e-tests/portal/cypress/support/portal/createBssApplicationReadyForCheck.js]: Have updated all instances to use `cy.keyboardInput`.

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/filter-by/dashboard-deals-filter-by-status.spec.js [hunk 23]
```diff
@@ -200,7 +203,7 @@ context('Dashboard Deals filters - filter by status', () => {
     });
 
     it('renders all deals regardless of status', () => {
-      dashboardDeals.rows().should('have.length', ALL_DEALS.length);
+      dashboardDeals.rows().should('have.length', 3);
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/filter-by/dashboard-deals-filter-by-status.spec.js:206]: Would be better to use a const here rather than `3`

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/filter-by/dashboard-deals-filter-by-status.spec.js [hunk 24]
```diff
@@ -4,22 +4,31 @@ const CONSTANTS = require('../../../../../fixtures/constants');
...
+      dealType: CONSTANTS.DEALS.DEAL_TYPE.BSS_EWCS,
+      status: CONSTANTS.DEALS.DEAL_STATUS.DRAFT,
+      // other deal properties...
+    },
+    {
+      dealType: CONSTANTS.DEALS.DEAL_TYPE.BSS_EWCS,
+      status: CONSTANTS.DEALS.DEAL_STATUS.READY_FOR_APPROVAL,
+      // other deal statuses...
+    },
+  ];
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/filter-by/dashboard-deals-filter-by-status.spec.js:25]: Maybe should revert the changes to this file, not sure why we have 2x BSS deals in `ALL_DEALS` which doesn't seem to be used and then we only call `cy.createBssEwcsDeal` once

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/filters/dashboard-deals-panel-filter-remove.spec.js [hunk 25]
```diff
@@ -15,7 +15,7 @@ context('Dashboard Deals - panel selected filters - remove a filter', () => {
     cy.deleteGefApplications(ADMIN);
     cy.deleteDeals(ADMIN);
 
-    cy.insertOneDeal(BSS_DEAL_DRAFT, BANK1_MAKER1).then((deal) => {
+    cy.createBssEwcsDeal({ readyForCheck: true }).then((deal) => {
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/filters/dashboard-deals-panel-filter-remove.spec.js:18]: This won't work as `createBssEwcsDeal` does not return a deal

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-facilities/filters/dashboard-facilities-filters.spec.js [hunk 26]
```diff
@@ -15,7 +15,9 @@ context('Dashboard Deals filters', () => {
     cy.deleteGefApplications(ADMIN);
     cy.deleteDeals(ADMIN);
 
-    cy.insertOneDeal(BSS_DEAL_DRAFT, BANK1_MAKER1);
+    cy.createBssEwcsDeal({ readyForCheck: true }).then((deal) => {
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-facilities/filters/dashboard-facilities-filters.spec.js:18]: This won't work as createBssEwcsDeal does not return a deal

File: e2e-tests/portal/cypress/e2e/pages/eligibilityDocumentation.js [hunk 27]
```diff
@@ -15,6 +15,7 @@ const page = {
   financialForecast: () => cy.get('[data-cy="financialForecasts"]'),
   financialCommentary: () => cy.get('[data-cy="financialInformationCommentary"]'),
   corporateStructure: () => cy.get('[data-cy="corporateStructure"]'),
+  saveAndGoBack: () => cy.get('[data-cy="save-go-back-button"]'),
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/pages/eligibilityDocumentation.js:18]: Why has this been added, we have a partial and command for this

File: e2e-tests/portal/cypress/support/portal/loginGoToDealPage.js [hunk 28]
```diff
@@ -1,6 +1,6 @@
 const pages = require('../../e2e/pages');
 
-module.exports = (user, deal) => {
+module.exports = (user) => {
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/support/portal/loginGoToDealPage.js:3]: Need to add some documentation

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/filter-by/dashboard-deals-filter-by-keyword.spec.js [hunk 29]
```diff
@@ -87,10 +78,6 @@ context('Dashboard Deals filters - filter by keyword', () => {
     it(`renders only deals that have ${MOCK_KEYWORD} in a field`, () => {
       const ALL_KEYWORD_DEALS = ALL_DEALS.filter(({ exporter }) => exporter.companyName === MOCK_KEYWORD);
       dashboardDeals.rows().should('have.length', ALL_KEYWORD_DEALS.length);
-
-      const firstDraftDeal = ALL_KEYWORD_DEALS[0];
-
-      dashboardDeals.row.exporter(firstDraftDeal._id).should('have.text', MOCK_KEYWORD);
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/filter-by/dashboard-deals-filter-by-keyword.spec.js:93]: Need to revert the removal of this and update.

**Comment by ttbarnes** [comment]
Closing this - have created a new ticket, branch and PR - DTFS2-7431, https://github.com/UK-Export-Finance/dtfs2/pull/3594
